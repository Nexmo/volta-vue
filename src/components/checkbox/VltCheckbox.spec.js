import { shallowMount } from '@vue/test-utils';
import { VltCheckbox } from '..';

describe('vlt-checkbox', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltCheckbox, {
      propsData: {
        label: 'Is test ok?',
      },
    });
  });

  test('renders correctly with label "Is test ok?"', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('has a "Hello World" tooltip', () => {
    wrapper.setProps({
      label: 'Is test ok?',
      tooltipTitle: 'Hello World',
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is disabled', () => {
    wrapper.setProps({
      label: 'Is test ok?',
      disabled: true,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
