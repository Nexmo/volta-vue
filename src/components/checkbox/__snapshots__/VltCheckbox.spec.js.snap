// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-checkbox has a "Hello World" tooltip 1`] = `
<div
  class="Vlt-checkbox"
>
  <label>
    <span
      class="Vlt-checkbox__button"
    >
      <input
        type="checkbox"
      />
      <span
        class="Vlt-checkbox__icon"
      />
    </span>
     Is test ok? 
    <span
      class="Vlt-tooltip"
    >
      
      <svg
        class="Vlt-icon Vlt-icon--small Vlt-blue"
      >
        <use
          xlink:href="#V_Vlt-icon-help-negative"
        />
      </svg>
      
    </span>
  </label>
</div>
`;

exports[`vlt-checkbox is disabled 1`] = `
<div
  class="Vlt-checkbox"
>
  <label>
    <span
      class="Vlt-checkbox__button"
    >
      <input
        disabled=""
        type="checkbox"
      />
      <span
        class="Vlt-checkbox__icon"
      />
    </span>
     Is test ok? 
    <!--v-if-->
  </label>
</div>
`;

exports[`vlt-checkbox renders correctly with label "Is test ok?" 1`] = `
<div
  class="Vlt-checkbox"
>
  <label>
    <span
      class="Vlt-checkbox__button"
    >
      <input
        type="checkbox"
      />
      <span
        class="Vlt-checkbox__icon"
      />
    </span>
     Is test ok? 
    <!--v-if-->
  </label>
</div>
`;
